### ISSUE TITLE
Doubts with the Code/Method.

### ISSUE THREAD

**Issue by kulkarni-vishwanath** [KrishnaPriyaIITR/Women-Data-Science-Hackathon-by-Bain-Company#1]
Hi Krishna,

I have a few queries with respect to your code (methodology), if you could please explain it,

1. It wasn't mentioned in the problem statement that the data is from 2017 to 2019, how did you come to such a conclusion?
2. In your BAIN_KP_XGB_GB.ipynb notebook, you've created lags for 60,90,91 and 366 days, whereas in your BAIN_KP_RF_ET.ipynb notebook, the lags are for entirely different duration, could you please explain the rationale behind these different approaches?

Will be glad if you could answer these .

Thanks,
Vishwanath.

**Comment by krishnapriya-18** [comment]
> Hi Krishna,
> 
> I have a few queries with respect to your code (methodology), if you could please explain it,
> 
> 1. It wasn't mentioned in the problem statement that the data is from 2017 to 2019, how did you come to such a conclusion?
> 2. In your BAIN_KP_XGB_GB.ipynb notebook, you've created lags for 60,90,91 and 366 days, whereas in your BAIN_KP_RF_ET.ipynb notebook, the lags are for entirely different duration, could you please explain the rationale behind these different approaches?
> 
> Will be glad if you could answer these .
> 
> Thanks,
> Vishwanath.

Hi Vishwanath,

1. As I had to replace days with probable dates to make date-based features, so I investigated a bit about the company, It was a USA based company operating mostly in INDIA and USA, also public holidays column had 25 holidays approx per year. So these had to be national holidays not regional. After this information, I tried with few dates, but this particular setting allowed for the Public holidays' column correlated with 25th Dec of all years, 1st Jan, etc. So this was chosen.

2. In most cases, sales usually follow a weekly pattern, here it was two weekly initially then weekly.so and since the test data was of two months so lag had to start from 60 or 61 days or it would have nan values in the column. And after that, a lag of 7 was chosen so that it could capture the weekly pattern until 4 months. after than half-yearly lag and then one year and then 1.5 years. Now since I was predicting 600 courses separately, at one time the model had only 800 training data. So in the case of bagging models as it creates subsets of data it had enough data so all these lag features performed as expected. But for Boosting models it only had 800 rows for correcting errors, so what  I feel is more features led to bad scores, thus I had to select the important ones from the initial set of lags and that would obviously start from 60 and then 90, 91 for quarterly trend and 366 for yearly trend.

I tried my best to answer your questions. If you still have something in your mind. DO ASK.

**Comment by kulkarni-vishwanath** [comment]
Hi Krishna,

Thank you so much for your explanation, it is clear now. However, I have some more questions (sorry to bother you with questions),
1. I see that you didn't tune your hyperparameters at all and you've used all the default ones. Were you relying completely on the Features that you generated? 
2. This question is more generic, is it always better to rely on hyperparameter tuning or your feature generation? 

Thanks again.

**Comment by krishnapriya-18** [comment]
> Hi Krishna,
> 
> Thank you so much for your explanation, it is clear now. However, I have some more questions (sorry to bother you with questions),
> 
> 1. I see that you didn't tune your hyperparameters at all and you've used all the default ones. Were you relying completely on the Features that you generated?
> 2. This question is more generic, is it always better to rely on hyperparameter tuning or your feature generation?
> 
> Thanks again.

@kulkarni-vishwanath 

1. Yes, I did not tune hyperparameters because, this hackathon had no AV points so I wanted to focus on feature engineering part because hyperparameter tuning is known to all. This was just an illustration notebook for beginners, so I tried to keep it simple. This approach can be followed for any time series problem and would result in good score. Yes but tuning would obviously increase my rank by 5-6 ranks atleast.

2. Correct features can increase your scores by 10-20 % whereas parameter tuning increases score by not more than 1-10%. So it is always advisable to focus most of your time on feature engineering and last two days or one day of the hack in tuning. 

PS: numbers are relative not absolute. :). Also keep in mind these are wrt to machine learning algorithms. In Neural networks tuning plays a huge role.
